### PR TITLE
Fix 213

### PR DESCRIPTION
**Pull request by paulstapor**
Fixes the not correctly working option `start_indices` in visualize.parameters and covers it by a test

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/parameters.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5038717&bid=16595068)

File: pypesto/visualize/parameters.py [hunk 1]
```diff
@@ -244,6 +244,10 @@ def handle_inputs(result, free_indices_only, lb=None, ub=None,
 
     # parse indices which should be plotted
     if start_indices is not None:
+        # handle, if only a number was passed
+        if type(start_indices) in (float, int):
```
Comment (dweindl) [inline pypesto/visualize/parameters.py]: use `isinstance(x, numbers.Number)` to cover all numeric types

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/parameters.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5038717&bid=16595068)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/parameters.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5038717&bid=16595068)

**Comment by paulstapor** [comment]
Okay, seems like PEtab and amici_objective changes counteract something in visualize_optimizer_history.
These are the failing visualization tests. Beyond that, this PR should be fine and fixes indeed 213

**Comment by yannikschaelte** [comment]
yes, I am mildly aware that the petab tests currently break. I have hope to get that fixed and merged to develop today, then we can see here again ...

**Comment by paulstapor** [comment]
> yes, I am mildly aware that the petab tests currently break. I have hope to get that fixed and merged to develop today, then we can see here again ...

No pressure. :D
Just wanted to say it would be safe to merge this one. But if you prefer fixing the PEtab related stuff (which makes sense), totally fine. ;)

**Comment by yannikschaelte** [comment]
@paulstapor the error was nothing with petab, but in optimizer_history a numpy function is applied to a pandas dataframe, and how these work together seems to have changed.

**Pull request merged**